tests/CommunityToolkit.Aspire.Hosting.Dapr.AzureExtensions.Tests/ResourceCreationTests.cs

```diff
@@ -0,0 +1,316 @@
+﻿using Aspire.Hosting;
+using Aspire.Hosting.Utils;
+using Azure.Provisioning;
+using Azure.Provisioning.KeyVault;
+
+namespace CommunityToolkit.Aspire.Hosting.Dapr.AzureExtensions.Tests;
+
+public class ResourceCreationTests
+{
+    [Fact]
+    public void AddAzureDaprResource_AddsToAppBuilder()
+    {
+        var builder = DistributedApplication.CreateBuilder();
+
+        var daprStateBuilder = builder.AddDaprStateStore("daprState")
+                                      .AddAzureDaprResource("AzureDaprResource", _ =>
+                                      {
+                                          // no-op
+                                      });
+
+        using var app = builder.Build();
+
+        var appModel = app.Services.GetRequiredService<DistributedApplicationModel>();
+
+        var resource = appModel.Resources.OfType<AzureDaprComponentResource>().SingleOrDefault();
+
+        Assert.NotNull(resource);
```

Comment:
This can be replaced with `Assert.Single`

Reply:
done